shipyard-templates/shipyard_templates/etl.py

```diff
@@ -1,4 +1,17 @@
 from abc import ABC, abstractmethod
+from .exit_code_exception import ExitCodeException
+
+
+class UnauthorizedError(ExitCodeException):
+    def __init__(self, message: str):
+        self.message = f"Failed to connect to the API. Ensure that the API key provided is correct. Message from the API: {message}"
```

Comment:
small tweak here:
instead of `API key` you should say credentials to cast a wider net